library/src/main/java/com/otaliastudios/zoom/OverPanRangeProvider.kt

```diff
@@ -0,0 +1,20 @@
+package com.otaliastudios.zoom
+
+/**
+ * Defines the allowed range for overpan.
+ */
+interface OverPanRangeProvider {
+
+    /**
+     * @return the maximum horizontal overpan to allow
+     */
+    @ZoomApi.ScaledPan
+    fun getHorizontalOverPanRange(engine: ZoomEngine): Float
+
+    /**
+     * @return the maximum vertical overpan to allow
+     */
+    @ZoomApi.ScaledPan
+    fun getVerticalOverPanRange(engine: ZoomEngine): Float
```

Comment:
I would probably prefer if we had just one function with a boolean parameter. More handy for people who want to use the same value for horizontal/vertical (the majority I think).